src/Tizen.NUI/src/internal/WebView/WebHttpRequestInterceptor.cs

```diff
@@ -46,56 +52,174 @@ public string Url
+        /// To call it, application should have full response body ready for the intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// After this call, any further call on WebHttpRequestInterceptor results in undefined behavior.
         /// <param name="body">Contents of response</param>
-        /// <param name="length">Length of Contents of response</param>
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public bool AddResponseBody(string body, uint length)
+        public bool SetResponseBody(string body)
+        {
+            if (body == null)
+                return false;
+            bool result = Interop.WebHttpRequestInterceptor.AddResponseBody(interceptorHandle, body, (uint)body.Length);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return result;
+        }
+
+        /// <summary>
+        /// Writes whole response body with headers at once.
+        /// To call it, application should have full response headers and body ready for the intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// After this call, any further call on WebHttpRequestInterceptor results in undefined behavior.
+        /// <param name="headers">Headers of response</param>
+        /// <param name="body">Contents of response</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool SetResponse(string headers, string body)
+        {
+            if (body == null)
+                return false;
+            bool result = Interop.WebHttpRequestInterceptor.AddResponse(interceptorHandle, headers, body, (uint)body.Length);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return result;
+        }
+
+        /// <summary>
+        /// Writes a part of response body.
+        /// This function can be called only OUTSIDE WebContext.HttpRequestIntercepted.
+        /// If it returns false, handling the request is done.
+        /// Any further calls result in undefined behavior.
+        /// User should always check return value, because response to this request might not be needed any more,
+        /// and the function can return false even though user still has data to write.
+        /// <param name="chunk">Chunk of response</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool WriteResponseChunk(string chunk)
         {
-            bool result = Interop.WebHttpRequestInterceptor.AddResponseBody(SwigCPtr, body, length);
+            if (chunk == null)
+                return false;
+            bool result = Interop.WebHttpRequestInterceptor.WriteResponseChunk(interceptorHandle, chunk, (uint)chunk.Length);
```

Comment:
We could be set null as chunk, it means completes of response

```
 * @post After writing full response body in chunks using this function, call
 *       it again with NULL as @a chunk and 0 as @a length, to signal that
 *       response body is finished.
```

Reply:
Done.